spec/requests/api/v1/targets/match_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'Match Targets', type: :request do
```

Comment:
```suggestion
describe 'POST Targets', type: :request do
```
Here you are testing the POST endpoint for Target creation, then you will have a case for when there are matched targets and also for there not are matched targets (they should be context below)